include/LightGBM/dataset.h

```diff
@@ -710,6 +762,11 @@ class Dataset {
   bool use_missing_;
   bool zero_as_missing_;
   std::vector<int> feature_need_push_zeros_;
+  std::vector<std::vector<double>> raw_data_;
```

Comment:
Instead of raw float data, is that possible to use the bucketed value (map from int to bucketed float value) for linear function?

Reply:
Yes, I had this idea as well and tried it out. The accuracy was not as good. Also I think that the result could be very sensitive to outliers in the first and last bins which will change the bin average a lot. Using float instead of double is a good idea, I will give that a try.